apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs

```diff
@@ -28,8 +28,20 @@ namespace Google.Cloud.Diagnostics.Common
     /// </summary>
     internal sealed class SimpleManagedTracer : IManagedTracer
     {
+        /// <summary>
+        /// A class that represents a running trace span.
+        /// </summary>
+        private class Span : IDisposable
+        {
+            private readonly SimpleManagedTracer _tracer;
+            public Span(SimpleManagedTracer tracer) { _tracer = tracer; }
+
+            /// <summary> Ends the current span.</summary>
+            public void Dispose() =>_tracer.EndSpan();
+        }
+
         /// <summary>A mutex to protect the rate limiter instance.</summary>
-        private static object _stackLock = new object();
+        private readonly object _stackLock = new object();
```

Comment:
Definitely approve of this being readonly - being non-static is a really big change. Is that deliberate? My guess is that it is, but I thought I'd check. (Also, should the summary comment be updated?)